docs/formatV3.txt

```diff
@@ -591,6 +595,88 @@ These TOTP parameters fields have the following format:
                       default is 0. The format is that of a 40 bit (5 byte)
                       Password Safe time_t value as specified in Section 3.1.3.
 
+3.4 Attachments
+
+3.4.1 Attachment record fields
+
+An Attachment record has the following fields:
+
+Name                        Value        Type     Mandatory       Comments
+--------------------------------------------------------------------------
+AttUUID                     0x60        UUID          Y              [1]
+Title                       0x61        Text          N              [2]
+Creation Time               0x62        time_t        Y              [3]
+MediaType                   0x63        Text          Y              [4]
+FileName                    0x64        Text          N              [5]
+FilePath                    0x65        Text          N              [6]
+FileCreationTime            0x66        time_t        N              [7]
+FileModificationTime        0x67        time_t        N              [7]
+FileAccessTime              0x68        time_t        N              [7]
```

Comment:
For this feature and v4, what are the rules for updating the access time?  Having to resave the entire file each time a user, say, views an attachment seems like a lot (thinking of Android and syncing files).  

The file creation, mod, and access times may be difficult to know on Android other than 'now'.  The attachments won't be real files but URLs and data streams provided by the system.

Reply:
The fields are marked non mandatory, so maybe just ignoring those?

Doesn't Android provide the modification time? It's been a while since I touched some Android code.. From user-perspective, I think that's the most interesting one to have.

Reply:
I'll either ignore or set to 'now' whenever the attachment is added or modified.  Maybe I could keep an access time internally and write only if the file is written otherwise.

Android recently has disabled direct OS file access except for apps with special all files permissions.  The standard C/Java File APIs no longer work, and you have to use their custom storage framework.  Maybe there is some time info in it, I'd have to check.  I do know that there is nothing like a file path available only a file name and probably a MIME type.

Reply:
Or do like Linux does with the relatime filesystem option to update the access time only if the file has been modified since the previous access.

Reply:
Personally I never found atime very useful, most of the time it was causing performance issues on high I/O servers so I just disabled it.

Looking at the current code, it looks like not being updated by the existing clients either

 % grep -irl SetFileATime .
./src/core/ItemAtt.h